docker/Dockerfile.petsc

```diff
@@ -0,0 +1,121 @@
+# DockerFile for an environment into which firedrake can be installed. Based off
+# https://github.com/firedrakeproject/firedrake/blob/master/docker/Dockerfile.env
+
+FROM ubuntu:24.04
+
+# Update and install required packages for Firedrake
+USER root
+SHELL ["/bin/bash", "-o", "pipefail", "-c"]
+RUN apt-get update \
+    && apt-get -y dist-upgrade \
+    && DEBIAN_FRONTEND=noninteractive apt-get -y install tzdata \
+    && apt-get -y install curl vim docker.io openssh-client build-essential autoconf automake \
+        cmake gfortran git libopenblas-serial-dev libtool python3-dev python3-pip python3-tk \
+        python3-venv python3-requests zlib1g-dev libboost-dev sudo gmsh bison flex ninja-build \
+        libocct-ocaf-dev libocct-data-exchange-dev swig graphviz libcurl4-openssl-dev libxml2-dev \
+    && rm -rf /var/lib/apt/lists/*
+
+# Use a more sane locale
+ENV LC_ALL C.UTF-8
+
+# Change the `ubuntu` user to `firedrake`
+# and ensure that we do not run as root on self-hosted systems
+RUN usermod -d /home/firedrake -m ubuntu \
+    && usermod -l firedrake ubuntu \
+    && groupmod -n firedrake ubuntu \
+    && usermod -aG sudo firedrake \
+    && echo "firedrake:docker" | chpasswd \
+    && echo "firedrake ALL=(ALL) NOPASSWD: ALL" >>/etc/sudoers \
+    && ldconfig
+
+USER firedrake
+WORKDIR /home/firedrake
+
+# Fetch PETSc
+RUN git clone https://github.com/firedrakeproject/petsc.git
+
+# Build MPICH manually because we don't want PETSc to build it twice
+RUN bash -c 'export PACKAGES=/home/firedrake/petsc/packages; \
+    cd petsc; \
+    ./configure \
+        --COPTFLAGS=-O3 -march=native -mtune=native \
+        --CXXOPTFLAGS=-O3 -march=native -mtune=native \
+        --FOPTFLAGS=-O3 -march=native -mtune=native \
+        --with-c2html=0 \
+        --with-debugging=0 \
+        --with-fortran-bindings=0 \
+        --with-make-np=12 \
+        --with-shared-libraries=1 \
+        --with-zlib \
+        --download-eigen \
+        --download-fftw \
+        --download-hdf5 \
+        --download-hwloc \
+        --download-hypre \
+        --download-metis \
+        --download-mmg \
+        --download-mumps \
+        --download-mpich \
+        --download-mpich-device=ch3:sock \
+        --download-netcdf \
+        --download-parmmg \
+        --download-pastix \
+        --download-pnetcdf \
+        --download-ptscotch \
+        --download-scalapack \
+        --download-suitesparse \
+        --download-superlu_dist \
+        PETSC_ARCH=packages; \
+        mv ${PACKAGES}/include/petscconf.h ${PACKAGES}/include/old_petscconf.nope;'
+# Don't run make here, we only want MPICH and HWLOC
+# It is also necessary to move `petscconf.h` so packages isn't treated like a working PETSc
+
+# Build PETSc
```

Comment:
Seems to have installed without errors, with this printed at the end:
```
Now to check if the libraries are working do:
make PETSC_DIR=/home/firedrake/petsc PETSC_ARCH=default check
```
which I did and got an error
```
Running PETSc check examples to verify correct installation
Using PETSC_DIR=/home/firedrake/petsc and PETSC_ARCH=default
Possible error running C/C++ src/snes/tutorials/ex19 with 1 MPI process
See https://petsc.org/release/faq/
lid velocity = 0.0016, prandtl # = 1., grashof # = 1.
Number of SNES iterations = 2
Possible error running C/C++ src/snes/tutorials/ex19 with 2 MPI processes
See https://petsc.org/release/faq/
lid velocity = 0.0016, prandtl # = 1., grashof # = 1.
Number of SNES iterations = 2
Error! file1 check failed: output/ex19_hypre.out
/home/firedrake/petsc/src/snes/tutorials
Possible problem with ex19 running with HYPRE, diffs above
=========================================
Error! file1 check failed: output/ex19_fieldsplit_5.out
/home/firedrake/petsc/src/snes/tutorials
Possible problem with ex19 running with MUMPS, diffs above
=========================================
Error! file1 check failed: output/ex19_suitesparse.out
/home/firedrake/petsc/src/snes/tutorials
Possible problem with ex19 running with SuiteSparse, diffs above
=========================================
Error! file1 check failed: output/ex19_superlu.out
/home/firedrake/petsc/src/snes/tutorials
Possible problem with ex19 running with SuperLU_DIST, diffs above
=========================================
C/C++ example src/vec/vec/tests/ex47 run successfully with HDF5
Completed PETSc check examples
Error while running make check
gmake[1]: *** [makefile:131: check] Error 1
make: *** [GNUmakefile:17: check] Error 2
```

Reply:
Yeah that's because the Dockerfile deletes the test and example directories to save space.

Reply:
Ah sorry!